### ISSUE TITLE
Focus resets when navigating away from link inside scrollable container

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#5251]
**Reported by hhillen on 2015-07-29 07:09**
Normally, if a link or other focusable element is focused and the down arrow key is pressed, NVDA will start reading the next line from the link's location using browse mode. However, if the focused element is inside a scrollable container, pressing the down arrow key will cause keyboard focus to reset to the container itself. The user loses their position and will have to navigate back to where they were inside the scrollable container. 

To reproduce, use the [test case]([https://dl.dropboxusercontent.com/u/573324/testcases/bugreports/NVDA/focus_reset.html|)] provided (also attached). In this test case, "link 1" and "button 1" are placed outside the scrollable container and function as expected. "link 2, 3 and 4" and "button 2" are placed inside a scrollable container. Pressing the down arrow key while one of them is focused will cause focus to be set back on the scrollable container itself.

This only happens in the Firefox / NVDA profile, so please advice whether the solution should be provided by NVDA or by Mozilla.

**Comment by ehollig** [comment]
I can still reproduce this using Firefox 57 using the following STR:
1. Open the above HTML document in Firefox.
2. Press the tab key to focus "Link 3".
3. Press the down arrow key to read the next line in browse mode.
4. Press tab and notice focus is placed on "Link 2".

**Comment by Adriani90** [comment]
I can reproduce this issue in Firefox 64.0.2. However, I cannot see anything in the log. Could this be a bug in Firefox?
cc: @jcsteh

**Comment by jcsteh** [comment]
The original issue described by hhillen in https://github.com/nvaccess/nvda/issues/5251#issue-116051543 no longer exists. Pressing down arrow after link 2, button 2, link 3 or link 4 does reset the focus, but it does not reset the browse mode cursor.

The reason the focus gets reset is that in Firefox, scrollable containers are deliberately focusable so that they can be scrolled using the keyboard. (I seem to recall their being discussion of Chrome implementing this too, but I don't have a reference for that.) When you move the browse mode cursor, NVDA sets focus to the nearest focusable ancestor. In this case, that's the container.

I can confirm the issue concerning the tab key described by @ehollig in https://github.com/nvaccess/nvda/issues/5251#issuecomment-350089087. That occurs because NVDA's tab override code sees that the cursor is within the focus (the scrollable section), so it passes tab through to the browser rather than overriding it based on the location of the browse mode cursor. The only fix I can really think of here is to have tab override ignore ROLE_SECTION when making this decision.

**Comment by Adriani90** [comment]
Closing as fixed by #11190.
